backend-project/small_eod/search/grammar.py

```diff
@@ -0,0 +1,115 @@
+from pyparsing import (
+    Word,
+    Suppress,
+    CaselessLiteral,
+    infixNotation,
+    pyparsing_unicode,
+    removeQuotes,
+    opAssoc,
+    FollowedBy,
+    quotedString,
+)
+
+
+printables = pyparsing_unicode.printables
```

Comment:
```suggestion
printables = pyparsing_unicode.printables
```
Taki stricte estetyczny, niezobowiazujacy komentarz, moze `form pyparsinh.pyparsing_unicode import printables, alphas, nums, alphanums`

Reply:
Nie ma takiej możliwości, próbowałem, to nie są obiekty, a pola. Zobacz na https://github.com/pyparsing/pyparsing/blob/master/pyparsing_archive.py#L7378 .

Reply:
Rozumiem. W dosc specyficzny sposob ten pyparse definiuje subsety dla roznych jezykow. Bardzo ogolnie ciekawa biblioteka. Chetnie sprobuje przesledzic logike Twojego kodu

Reply:
W zakresie pyparsing dostrzegam, że przydałoby się, aby domyślnie wprowadzał `AND` pomiędzy dwa słowa, ale nie udało mi się tego osiągnąć, nie położyłem dostatecznego wysiłku, a domyślny `AND` byłbym ułatwieniem. Podpowiadam, bo może wpadniesz na sprytne rozwiązanie, gdy będziesz analizować.

Reply:
Optymalizacja, bo samo odwołanie do kilku teoretycznie statycznych ciągów znakowych kosztuje 1 sekundę. Przyjmuje, że to inicjalizacja, a przede wszystkim optymalizacja, więc się jeszcze tym nie martwię.

```
$ python -c 'import datetime; start=datetime.datetime.now(); from pyparsing import pyparsing_unicode; pyparsing_unicode.printables; pyparsing_unicode.alphas; pyparsing_unicode.nums; pyparsing_unicode.alphanums; print(datetime.datetime.now()-start);';
```